sr_utilities_common/scripts/sr_utilities_common/performance_test.py

```diff
@@ -0,0 +1,109 @@
+
+        rospy.Subscriber("/diagnostics_agg", DiagnosticArray, self._diagnostics_agg_callback)
+        if self._hand_e:
+            rospy.Subscriber("/rh/debug_etherCAT_data", EthercatDebug, self._ethercat_data_callback)
+
+        self.total_control_loop_overruns_count = 0
+        self.total_recent_control_loop_overruns_count = 0
+        self.total_invalid_packets_total_count = list()
+        self.total_invalid_packets_recent = 0
+        self._invalid_hand_e_packets_total = 0
+
+    def _diagnostics_agg_callback(self, diagnostics_full_msg):
+        diagnostics_control_loop_msg = None
+        diagnostics_ethercat_msg_list = list()
+
+        for msg in diagnostics_full_msg.status:
+            if msg.name == "/Realtime Control Loop/Realtime Control Loop":
+                diagnostics_control_loop_msg = msg
+
+            if msg.name.startswith("/EtherCat/EtherCAT Slaves/H0_"):
+                diagnostics_ethercat_msg_list.append(msg)
+
+        self._set_control_loop_values(diagnostics_control_loop_msg)
+        if self._hand_h:
+            self._set_ethercat_values(diagnostics_ethercat_msg_list)
+
+    def _ethercat_data_callback(self, msg):
+        msg_list = list()
+        msg_list.append(msg)
+        self._set_ethercat_values(msg_list)
+
+    def _set_ethercat_values(self, diagnostics_ethercat_msg_list):
+        invalid_packets_total_count_list = list()
+        for diagnostics_ethercat_msg in diagnostics_ethercat_msg_list:
+            if self._hand_h:
+                for value_list in diagnostics_ethercat_msg.values:
+                    if value_list.key == "Invalid Packets Total":
+                        invalid_packets_total_count_list.append(float(value_list.value))
+                    if value_list.key == "Invalid Packets Recent":
+                        self.total_invalid_packets_recent += float(value_list.value)
+                self.total_invalid_packets_total_count = deepcopy(invalid_packets_total_count_list)
+            elif self._hand_e:
+                if diagnostics_ethercat_msg.motor_data_type.data == 0:
+                    self._invalid_hand_e_packets_total += 1
+
+    def _set_control_loop_values(self, diagnostics_control_loop_msg):
+        for value_list in diagnostics_control_loop_msg.values:
+            if value_list.key == "Control Loop Overruns":
+                control_loop_overruns_count = value_list.value
+            if value_list.key == "Recent Control Loop Overruns":
+                recent_control_loop_overruns_count = value_list.value
+
+        self.total_control_loop_overruns_count = float(control_loop_overruns_count)
+        self.total_recent_control_loop_overruns_count += float(recent_control_loop_overruns_count)
+
+    def _print_results(self, duration):
+        avg_recent_control_loop_overruns_count = self.total_recent_control_loop_overruns_count / duration
+        avg_total_invalid_packets_recent = self.total_invalid_packets_recent / duration
```

Comment:
`self.total_invalid_packets_recent` should not exist. that's why we have `self.total_invalid_packets_total_count`
The "recent" values should not be averaged (not for the control loop overruns either).
For the "recent" values we only need to check the maximum (i.e. what were the worst figures that we got in a certain 1 second period).